platform/ios/src/MGLMapView.mm

```diff
@@ -3794,8 +3777,12 @@ - (MGLAnnotationView *)annotationViewForAnnotation:(id<MGLAnnotation>)annotation
         annotationView.mapView = self;
         CGRect bounds = UIEdgeInsetsInsetRect({ CGPointZero, annotationView.frame.size }, annotationView.alignmentRectInsets);
 
-        _largestAnnotationViewSize = CGSizeMake(MAX(_largestAnnotationViewSize.width, CGRectGetWidth(bounds)),
-                                                MAX(_largestAnnotationViewSize.height, CGRectGetHeight(bounds)));
+        // Take any offset into consideration
+        CGFloat adjustedAnnotationWidth = CGRectGetWidth(bounds) + fabs(annotationView.centerOffset.dx);
```

Comment:
The annotation's apparent size now includes any offset, so that the initial annotation query returns the correct results. This means that `_unionedAnnotationRepresentationSize` can now be larger. This could be a performance issue with lots of annotations if the offsets are large.